### PR TITLE
Rewrite/add template-related unit tests

### LINKED ISSUE #268: graphite-output unit tests
The graphite-output template function needs some unit tests.

### PR DESCRIPTION
**Pull request by MrAnno**
This PR
- adds unit tests for the `$(graphite-output)` template function
- rewrites all template-related unit tests in Criterion

Resolves #268

### CODE REVIEW

**Comment by kira-syslogng** [comment]
Build SUCCESS, the tests were executed on test branch: master and test suite: functions

**Comment by kira-syslogng** [comment]
Build SUCCESS, the tests were executed on test branch: master and test suite: functions

**Comment by kira-syslogng** [comment]
Build SUCCESS, the tests were executed on test branch: master and test suite: functions

File: libtest/cr_template.h [hunk 1]
```diff
@@ -22,13 +22,16 @@
  *
  */
 
-#ifndef LIBTEST_TEMPLATE_LIB_H_INCLUDED
-#define LIBTEST_TEMPLATE_LIB_H_INCLUDED 1
+#ifndef LIBTEST_CR_TEMPLATE_H_INCLUDED
+#define LIBTEST_CR_TEMPLATE_H_INCLUDED 1
 
-#include "testutils.h"
+#include "syslog-ng.h"
```
Comment (furiel) [inline libtest/cr_template.h]: Why do we need to include syslog-ng.h?

**Comment by kira-syslogng** [comment]
Build SUCCESS, the tests were executed on test branch: master and test suite: functions

File: modules/stardate/tests/test_stardate.c [hunk 2]
```diff
@@ -20,16 +20,21 @@
...
+#include <criterion/criterion.h>
+
 #include <syslog-ng.h>
 #include <logmsg/logmsg.h>
-#include <template_lib.h>
+#include <libtest/cr_template.h>
 #include <apphook.h>
 #include <plugin.h>
 
+<<<<<<< HEAD
```
Comment (furiel) [inline modules/stardate/tests/test_stardate.c]: Missing resolve?

Comment (MrAnno) [reply modules/stardate/tests/test_stardate.c]: Well.. Sorry, I forgot to press "save"..

**Comment by kira-syslogng** [comment]
success

**Comment by kira-syslogng** [comment]
Build SUCCESS, the tests were executed on test branch: master and test suite: functions

**Comment by furiel** [comment]
I resolved the conflicts and aligned the patchset to #1878. Ready to review.

**Comment by MrAnno** [comment]
Nice, thank you very much! :+1:

File: lib/template/tests/test_template.c [hunk 3]
```diff
@@ -105,8 +107,33 @@ assert_template_format_multi_thread(const gchar *template, const gchar *expected
...
+{
+  app_startup();
+
+  init_template_tests();
+  cfg_load_module(configuration, "basicfuncs");
+  configuration->template_options.frac_digits = 3;
+  configuration->template_options.time_zone_info[LTZ_LOCAL] = time_zone_info_new(NULL);
+
+
+  putenv("TZ=MET-1METDST");
```
[the remark this answers is not in the document]
Comment (furiel) [reply lib/template/tests/test_template.c]: Thanks for the notice. Fixed.

File: libtest/cr_template.c [hunk 4]
```diff
@@ -22,13 +22,33 @@
  *
  */
 
-#include "testutils.h"
-#include "template_lib.h"
-#include "msg_parse_lib.h"
+#include <criterion/criterion.h>
+
+#include "cr_template.h"
```
[the remark this answers is not in the document]
Comment (furiel) [reply libtest/cr_template.c]: Fixed.

[the remark this answers is not in the document]
Comment (MrAnno) [reply libtest/cr_template.c]: @furiel Thanks again for finishing this PR :)

**Comment by kira-syslogng** [comment]
success

File: libtest/cr_template.h (hunk 1 above)
Comment (gaborznagy) [reply libtest/cr_template.h]: I think to show dependencies explicitly: syslog-ng.h contains definition of LogMessage.
Of course other headers, e.g. templates.h includes it.

File: libtest/cr_template.c [hunk 5]
```diff
@@ -39,8 +58,6 @@ void
 deinit_template_tests(void)
 {
   deinit_syslogformat_module();
-  if (!testutils_deinit())
```
Comment (gaborznagy) [inline libtest/cr_template.c]: This was the last user of `testutils_deinit`.
As we try to move on to criterion from libtest: should we delete those funcs one-by-on from testutils or we will remove everything in one step when the last testcase depending on testutils is removed?

**Pull request merged**